lib/atomic_cache/atomic_cache_client.rb

```diff
@@ -95,8 +103,6 @@ def generate_and_store(keyspace, options, tags)
           return nil
         end
```

Comment:
Moving new key up creates the issue of preserving a last_known_key to restore to if we end up here, then deleting the promoted value, maybe? This case of needing a key available to other threads before a value is created is tricky.